lib/logstash/inputs/elasticsearch.rb

```diff
@@ -63,12 +66,16 @@
 #
 #
 class LogStash::Inputs::Elasticsearch < LogStash::Inputs::Base
+
+  include LogStash::PluginMixins::ECSCompatibilitySupport(:disabled, :v1, :v8 => :v1)
+  include LogStash::PluginMixins::ECSCompatibilitySupport::TargetCheck
+
+  include LogStash::PluginMixins::EventSupport::EventFactoryAdapter
+
   extend LogStash::PluginMixins::ValidatorSupport::FieldReferenceValidationAdapter
 
   config_name "elasticsearch"
 
-  default :codec, "json"
```

Comment:
Side note: I think we need a way to specify a noop codec. Because the base classes default to `codec => plain`, we instantiate a codec that we will never use during setup of the plugin.

Reply:
:+1: here the reason for removal is mostly due json codec warning in ECS mode on an unset `target`